bottles/frontend/ui/dialog-fsr.blp

```diff
@@ -0,0 +1,68 @@
+using Gtk 4.0;
+
+template FsrDialog : .AdwWindow {
+  default-width: "500";
+  modal: true;
+
+  title: _("FidelityFX Super Resolution Settings");
+
+  ShortcutController {
+    Shortcut {
+      trigger: "Escape";
+      action: "action(window.close)";
+    }
+  }
+
+  Box {
+    orientation: vertical;
+
+    .AdwHeaderBar {
+      show-start-title-buttons: false;
+      show-end-title-buttons: false;
+
+      Button {
+        label: _("_Cancel");
+        use-underline: true;
+        action-name: "window.close";
+      }
+
+      [end]
+      Button btn_save {
+        label: _("Save");
+
+        styles [
+          "suggested-action",
+        ]
+      }
+    }
+
+    .AdwPreferencesPage {
+
+      .AdwPreferencesGroup {
+        .AdwComboRow combo_upscaling_resolution_mode {
+          title: "Upscaling Resolution Mode";
```

Comment:
What does this mean?
Sounds really awkward.

Reply:
You're right. It's essentially FSR modes; the higher the quality, the lower the performance. What are your thoughts on "Quality Mode"? That's how AMD calls it.